lib/core/CJsonLogLayout.cc

```diff
@@ -63,132 +59,103 @@ bool CJsonLogLayout::locationInfo() const {
     rapidjson::StringBuffer buffer;
-    TStringBufferWriter writer(buffer);
+    TStringBufferWriter writer{buffer};
 
     writer.StartObject();
 
     writer.String(LOGGER_NAME);
-    LOG4CXX_ENCODE_CHAR(logger, event->getLoggerName());
-    writer.String(logger);
+    writer.String(LOGGER);
 
     writer.String(TIMESTAMP_NAME);
-    writer.Int64(event->getTimeStamp() / 1000);
+    const boost::posix_time::ptime& timeStamp{
+        boost::log::extract<boost::posix_time::ptime>(
+            boost::log::aux::default_attribute_names::timestamp(), rec)
+            .get()};
+    writer.Int64((timeStamp - EPOCH).total_milliseconds());
 
     writer.String(LEVEL_NAME);
-    LOG4CXX_ENCODE_CHAR(level, event->getLevel()->toString());
-    writer.String(level);
+    CLogger::ELevel level{boost::log::extract<CLogger::ELevel>(
+                              boost::log::aux::default_attribute_names::severity(), rec)
+                              .get()};
+    writer.String(CLogger::levelToString(level));
 
     writer.String(PID_NAME);
     writer.Int64(PID);
 
     writer.String(THREAD_NAME);
-    LOG4CXX_ENCODE_CHAR(thread, event->getThreadName());
-    writer.String(thread);
+    const boost::log::attributes::current_thread_id::value_type threadId{
+        boost::log::extract<boost::log::attributes::current_thread_id::value_type>(
+            boost::log::aux::default_attribute_names::thread_id(), rec)
+            .get()};
+    std::ostringstream oss;
+    oss << threadId;
+    writer.String(oss.str());
 
     writer.String(MESSAGE_NAME);
-    LOG4CXX_ENCODE_CHAR(message, event->getRenderedMessage());
+    const std::string& message{rec[boost::log::expressions::smessage].get()};
     writer.String(message);
 
-    LogString logNdc;
-    if (event->getNDC(logNdc)) {
-        writer.String(NDC_NAME);
-        LOG4CXX_ENCODE_CHAR(ndc, logNdc);
-        writer.String(ndc);
-    }
-
     if (m_LocationInfo) {
-        const spi::LocationInfo& locInfo = event->getLocationInformation();
 
-        const std::string& className = locInfo.getClassName();
-        if (!className.empty()) {
+        std::string functionName{boost::log::extract<std::string>(
+                                     CLogger::instance().functionAttributeName(), rec)
+                                     .get()};
+
+        // Example:
+        //
+        // Function = std::string ns1::ns2::clazz::someMethod(int arg1, char arg2)
+        //
+        // gets mapped to:
+        //
+        // Class = ns1::ns2::clazz
+        // Method = someMethod
+
+        std::size_t argsStartPos{functionName.find('(')};
+        if (argsStartPos != std::string::npos) {
+            functionName.erase(argsStartPos);
+        }
+        std::size_t returnTypeSeparatorPos{functionName.rfind(' ')};
+        if (returnTypeSeparatorPos != std::string::npos) {
+            functionName.erase(0, returnTypeSeparatorPos + 1);
+        }
+        std::size_t lastScopeOperatorPos{functionName.rfind("::")};
```

Comment:
Maybe factor out into cc local method or `CJsonLogLayout` helper for readability.